dev/Repeater/ItemsRepeater.cpp

```diff
@@ -110,7 +110,15 @@ winrt::Size ItemsRepeater::MeasureOverride(winrt::Size const& availableSize)
 
     if (auto layout = Layout())
     {
-        desiredSize = layout.Measure(GetLayoutContext(), availableSize);
+        auto layoutContext = GetLayoutContext();
+
+        // Expensive operation, do it only in debug builds.
+#ifdef _DEBUG
+        auto virtualContext = winrt::get_self<VirtualizingLayoutContext>(layoutContext);
+        virtualContext->Indent(Indent());
+#endif
```

Comment:
I won't block on this but this function would benefit from using a lamda to define your local variables, using this you could avoid the need for defining empty winrt::Size and winrt::Rect 

Something like:

winrt::Size desiredSize = [this]()
{ 
if (auto layout = Layout())
    {
        desiredSize = layout.Measure(GetLayoutContext(), availableSize);
        auto layoutContext = GetLayoutContext();

        // Expensive operation, do it only in debug builds.
#ifdef _DEBUG
        auto virtualContext = winrt::get_self<VirtualizingLayoutContext>(layoutContext);
        virtualContext->Indent(Indent());
#endif

        return layout.Measure(layoutContext, availableSize);
    }
return winrt::Size{0,0}
}();      

You could extend it to some of your other locals too:

auto layoutContext = [this]()
{
        auto layoutContext = GetLayoutContext();

        // Expensive operation, do it only in debug builds.
#ifdef _DEBUG
        auto virtualContext = winrt::get_self<VirtualizingLayoutContext>(layoutContext);
        virtualContext->Indent(Indent());
#endif
return layoutContext;
}();

With the right messaging you could get it so that you don't have to redefine your locals like:

    auto layout = Layout()